gdk/x11/gdkglcontext-x11-eglx.c

```diff
@@ -810,13 +843,28 @@ gdk_x11_display_make_gl_context_current (GdkDisplay   *display,
 
   if (context_x11->is_attached)
     {
-      /* If the WM is compositing there is no particular need to delay
-       * the swap when drawing on the offscreen, rendering to the screen
-       * happens later anyway, and its up to the compositor to sync that
-       * to the vblank. */
       GdkScreen *screen = gdk_window_get_screen (window);
+      gboolean is_composited = gdk_screen_is_composited (screen);
+
+#ifdef HAVE_XCOMPOSITE
+      GdkToplevelX11 *toplevel = _gdk_x11_window_get_toplevel (window);
+
+      /* If the WM is compositing and the window is redirected there is no
+       * particular need to delay the swap when drawing on the offscreen,
+       * rendering to the screen happens later anyway, and its up to the
+       * compositor to sync that to the vblank.
+       */
+      if (is_composited)
+        {
+          if (toplevel && gdk_x11_screen_supports_net_wm_hint (screen,
+              gdk_atom_intern_static_string ("_GTK_WINDOW_UNREDIRECTED")))
```

Comment:
Coding style: do not like break; if the line is too long, use a temporary boolean variable.